src/main/java/edu/greenblitz/tobyDetermined/Nodesssss/AStar.java

```diff
@@ -0,0 +1,100 @@
+package edu.greenblitz.tobyDetermined.Nodesssss;
+
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.Map;
+
+public class AStar {
+    private static final NodeBase nodeBase = NodeBase.getInstance();
+
+  public static double getDistanceFromCurrentNodeToEndOrStartNode(NodeArm point, NodeArm current){
+       return Math.abs(nodeBase.getDistanceBetweenToPoints(point,current));
+
+  }
+    public static double getDistanceToStartPlusEnd(NodeArm current, NodeArm start, NodeArm end){
+       double gCost = getDistanceFromCurrentNodeToEndOrStartNode(start,current);
+       double hCost = getDistanceFromCurrentNodeToEndOrStartNode(end,current);
+       return gCost+hCost;
+    }
+
+
+    public static NodeArm getLowestFcost(LinkedList<NodeArm> open, NodeArm start, NodeArm end) {
+       if (open.isEmpty()) {
+           return null; // Handle the case where the list is empty
+       }
+       int saveI = 0;
+       double fCost = getDistanceToStartPlusEnd(open.get(0), start, end);
+       for (int i = 1; i < open.size(); i++) {
+           double currentFCost = getDistanceToStartPlusEnd(open.get(i), start, end);
+           if (currentFCost < fCost) {
+               fCost = currentFCost;
+               saveI = i;
+           }
+       }
+       return open.get(saveI);
+   }
+
+
+   public static boolean isInList(NodeArm nodeArm, LinkedList<NodeArm> list){
+      for (NodeArm arm : list) {
```

Comment:
Not critical but consider looking into a build-in methods in Java's `Collection` class (which LinkedList should inherit from) to check this, probably something like `filter`

Reply:
changed to return list.contains(nodeArm)